arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/EagerEffect.kt

```diff
@@ -96,7 +96,7 @@ public interface EagerEffect<R, A> {
 
   public fun <B> map(f: (A) -> B): EagerEffect<R, B> = flatMap { a -> eagerEffect { f(a) } }
 
-  public fun <B> flatMap(f: (A) -> EagerEffect<R, B>): EagerEffect<R, B> = eagerEffect {
+  public fun <B> flatMap(f: (A) -> EagerEffect<@UnsafeVariance R, B>): EagerEffect<R, B> = eagerEffect {
```

Comment:
Damn, these were methods we removed for `Effect` but seem left over for `Effect`.

Reply:
should we remove flatMap?

Reply:
Maybe we should deprecate it... it probably has not been in use a lot anyway. cc\\ @raulraja @franciscodr

Reply:
I think there is no need for flatMap or any lambda taking combinators related to the functor, applicative, and monad hierarchies whenever there is the ability to `bind`. The smaller the API the better. For the scala 3 version of some of these abstractions, I'm avoiding implementing any of those until we need them.

Reply:
Agreed, so let's deprecate these methods so we avoid people using them.

Reply:
I'll add the deprecations it in this PR right away

Reply:
I am pushing this into a different branch for a cleaner git history